### PR TITLE
Add firewall dependency for gitea/nexus

### LINKED ISSUE #1297: First build for a PR fails with 'Error: Firewall "fw-***" was not found in Resource Group "rg-***"'
**Describe the bug**
The first build for a PR fails with `Error: Firewall "fw-***" was not found in Resource Group "rg-***"` for gitea and nexus:

```
╷
│ Error: Firewall "fw-***" was not found in Resource Group "rg-***"
│ 
│   with module.gitea[0].data.azurerm_firewall.fw,
│   on ../../shared_services/gitea/terraform/data.tf line 33, in data "azurerm_firewall" "fw":
│   33: data "azurerm_firewall" "fw" {
│ 
╵
╷
│ Error: Firewall "fw-***" was not found in Resource Group "rg-***"
│ 
│   with module.nexus[0].data.azurerm_firewall.fw,
│   on ../../shared_services/sonatype-nexus/terraform/data.tf line 38, in data "azurerm_firewall" "fw":
│   38: data "azurerm_firewall" "fw" {
│ 
╵
```

**Steps to reproduce**

1. Create a PR
1. add the `safe to test` label

### PR DESCRIPTION
**Pull request by stuartleeks**
Add dependency on `firewall` module to `gitea` and `nexus` modules to fix #1297

### CODE REVIEW

**Comment by tamirkamara** [comment]
@stuartleeks perhaps it would be nice to add the dependency for Gitea too?

**Review by martinpeck — APPROVED** [review]
LGTM

**Comment by martinpeck** [comment]
> @stuartleeks perhaps it would be nice to add the dependency for Gitea too?

how do you mean @tamirkamara ?

**Comment by stuartleeks** [comment]
> @stuartleeks perhaps it would be nice to add the dependency for Gitea too?

@tamirkamara  - do you mean something other than this?

image

https://github.com/microsoft/AzureTRE/blob/caab6bd33aa4c52acc7844f318cfc694647ffe25/templates/core/terraform/main.tf#L133

**Review by tamirkamara — APPROVED** [review]
I probably missed that line about gitea (not sure how).

File: .github/workflows/deploy_tre_reusable.yml [hunk 1]
```diff
@@ -54,13 +54,42 @@ on:
...
         required: true
+      GH_ACTIONS_DEBUG_PRIVATE_KEY:
+        required: false
 
 # This will prevent multiple runs of this entire workflow.
 # We should NOT cancel in progress runs as that can destabilize the environment.
 concurrency: "deploy-${{ github.ref }}"
 
 jobs:
+  debug:
```
Comment (ross-p-smith) [inline .github/workflows/deploy_tre_reusable.yml:65]: Assuming that this is not related to this PR?

File: .github/workflows/deploy_tre.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ on:
     # 1am each night https://crontab.guru/#0_1_*_*_*
     - cron: "0 1 * * *"
   push:
-    branches: [main]
+    # branches: [main]
```
Comment (ross-p-smith) [inline .github/workflows/deploy_tre.yml:9]: Assuming test

**Review by ross-p-smith — CHANGES_REQUESTED** [review]
Assuming you are using this for debugging something

**Comment by stuartleeks** [comment]
Closing and re-opening to test whether the build issue is transient or not

**Pull request closed without merging**